networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/reputation/ReputationManager.java

```diff
@@ -121,5 +141,17 @@ private boolean isLocallyConsideredUnsuitable(
       return locallyInitiated
           && reason.map(r -> !LOCAL_TEMPORARY_DISCONNECT_REASONS.contains(r)).orElse(false);
     }
+
+    public boolean adjustReputation(final ReputationAdjustment effect) {
+      final int newScore =
+          score.updateAndGet(
+              current -> Math.min(MAX_REPUTATION_SCORE, current + effect.getScoreChange()));
+      final boolean shouldDisconnect = newScore <= DISCONNECT_THRESHOLD;
+      if (shouldDisconnect) {
+        // Prevent our node from connecting out to the remote peer.
+        unsuitable = true;
+      }
```

Comment:
this peer is then unsuitable until restart, is that a bit harsh? should it just have a 'naughty corner' timeout?

Reply:
The peer can still connect to it, we just won't initiate a connection to the peer anymore.  Given they just tried to feed us bad info that seems wise. At some point we will likely need to go further and completely ban them, though that likely would be an even lower score (ie first time you just get disconnected, second time you get banned kind of policy).